libs-haskell/da-hs-base/src/Options/Applicative/Extended.hs

```diff
@@ -53,3 +56,24 @@ flagYesNoAuto flagName defaultValue helpText mods =
     determineAuto defaultValue <$> flagYesNoAuto' flagName (helpText <> commonHelp) mods
     where
         commonHelp = " Can be set to \"yes\", \"no\" or \"auto\" to select the default (" <> show defaultValue <> ")"
+
+-- | optparse-applicative does not provide useful error messages when a valid
+-- option is passed more than once https://github.com/pcapriotti/optparse-applicative/issues/395
+--
+-- This provides better error messages by constructing and running two parsers
+-- for the same option, where the second parser throws an error if it is
+-- invoked.
+--
+-- If the option is specified a second time, the parser is invoked and triggers
+-- an error.
+optionOnce :: ReadM a -> Mod OptionFields a -> Parser a
```

Comment:
Can you add a couple of unit tests for this behaviour? It's fine to use something like execParser like in the issue you referenced: https://github.com/pcapriotti/optparse-applicative/issues/395

Reply:
done